nexus/internal-api/src/lib.rs

```diff
@@ -0,0 +1,587 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+/// Path parameters for Sled Agent requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct SledAgentPathParam {
+    pub sled_id: Uuid,
+}
+
+/// Path parameters for Rack requests.
+#[derive(Deserialize, JsonSchema)]
+pub struct RackPathParam {
+    pub rack_id: Uuid,
+}
+
+/// Path parameters for Switch requests.
+#[derive(Deserialize, JsonSchema)]
+pub struct SwitchPathParam {
+    pub switch_id: Uuid,
+}
+
+/// Path parameters for Instance requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct InstancePathParam {
+    pub instance_id: Uuid,
+}
+
+/// Path parameters for Disk requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct DiskPathParam {
+    pub disk_id: Uuid,
+}
+
+/// Path parameters for Volume requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct VolumePathParam {
+    pub volume_id: Uuid,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize, JsonSchema, Serialize)]
+pub struct CollectorIdPathParams {
+    /// The ID of the oximeter collector.
+    pub collector_id: Uuid,
+}
+
+/// Path parameters for Upstairs requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct UpstairsPathParam {
+    pub upstairs_id: UpstairsUuid,
+}
+
+/// Path parameters for Upstairs requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct UpstairsRepairPathParam {
+    pub upstairs_id: UpstairsUuid,
+    pub repair_id: UpstairsRepairUuid,
+}
+
+/// Path parameters for Downstairs requests (internal API)
+#[derive(Deserialize, JsonSchema)]
+pub struct UpstairsDownstairsPathParam {
+    pub upstairs_id: UpstairsUuid,
+    pub downstairs_id: DownstairsUuid,
+}
+
+/// Path parameters for Saga requests
+#[derive(Deserialize, JsonSchema)]
+pub struct SagaPathParam {
+    pub saga_id: Uuid,
+}
+
+/// Path parameters for Background Task requests
+#[derive(Deserialize, JsonSchema)]
+pub struct BackgroundTaskPathParam {
+    pub bgtask_name: String,
+}
+
+/// Query parameters for Background Task activation requests.
+#[derive(Deserialize, JsonSchema)]
+pub struct BackgroundTasksActivateRequest {
+    pub bgtask_names: BTreeSet<String>,
+}
+
+/// Path parameters for NAT ChangeSet
+#[derive(Deserialize, JsonSchema)]
+pub struct RpwNatPathParam {
+    /// which change number to start generating
+    /// the change set from
+    pub from_gen: i64,
+}
+
+/// Query parameters for NAT ChangeSet
+#[derive(Deserialize, JsonSchema)]
+pub struct RpwNatQueryParam {
+    pub limit: u32,
+}
+
+/// Path parameters for probes
+#[derive(Deserialize, JsonSchema)]
+pub struct ProbePathParam {
+    pub sled: Uuid,
+}
+
+#[dropshot_server]
+pub trait NexusInternalApi: Send + Sync + Sized + 'static {
+    /// Return information about the given sled agent
+    #[endpoint {
+        method = GET,
+        path = "/sled-agents/{sled_id}",
+    }]
+    async fn sled_agent_get(
+        rqctx: RequestContext<Self>,
```

Comment:
Static dispatch is nicer because you don't have to prove that the trait is object-safe -- you can also add non-endpoint methods to the trait.